d2/src/main/java/com/linkedin/d2/balancer/subsetting/DeterministicSubsettingStrategy.java

```diff
@@ -82,42 +72,31 @@ public DeterministicSubsettingStrategy(DeterministicSubsettingMetadataProvider m
   }
 
   @Override
-  public boolean isSubsetChanged(long version)
+  public long getPeerClusterVersion()
```

Comment:
Related to the other question - I believe one strategy maps to one d2Service, but the peerClusterVersion is for the whole cluster, I'm not sure if it's the right place to duplicate the cluster version inside each strategy

Reply:
This is fine. We would like to listen to the cluster changes that the service belongs to. When the cluster version is updated, the URIs that serve this service are also expected to be updated

Reply:
Another idea I have is to have ZKDeterministicSubsettingMetadataProvider to directly update the subsetState in the SimpleLoadBalancerState. This way, each strategy doesn't have to interact with the peerVersion directly. Instead of pulling from ZKDeterministicSubsettingMetadataProvider every time from different service based strategy, we can push the update to  the state whenever we receive a new change. This is similar to what the uri listener does today.

Reply:
The current implementation is built using the same idea. Even though there's one strategy for each D2 service, there's only a single shared ZKDeterministicSubsettingMetadataProvider instance. We use the URIProperties from LoadBalancerState as the source of truth for the latest peer cluster version. The SubsettingState is used for storing the cache of service -> subset map. The _peerClusterVersion in SubsettingState is not necessarily the latest one, but the last cached one.